src/components/input-with-icon/InputWithIcon.tsx

```diff
@@ -0,0 +1,35 @@
+import { FC, ReactNode } from 'react'
+import IconButton from '@mui/material/IconButton'
+import ClearRoundedIcon from '@mui/icons-material/ClearRounded'
+import Box from '@mui/material/Box'
+import InputBase, { InputBaseProps } from '@mui/material/InputBase'
+import { SxProps } from '@mui/material'
+
+import { styles } from '~/components/input-with-icon/InputWithIcon.styles'
+
+interface InputWithIconProps extends InputBaseProps {
+  startIcon?: ReactNode
+  onClear: () => void
+}
+
+const InputWithIcon: FC<InputWithIconProps> = ({
+  value,
+  onClear,
+  startIcon,
+  sx,
+  ...props
+}) => {
+  return (
+    <Box sx={[styles.root, sx] as SxProps}>
```

Comment:
I guess, type Sx = { root: SxProps, input: SxProps, clearIcon: SxProps} or using an index signature can prevent you from using 'as'

Reply:
sx is not for {root,input} i use it only for this box, I tried solutions from mui docs but always got an error only as solved this